File: R/get_cached_data.R
#' get_dataset 
#' 
#' Download a predefined set of time series. 
#' @param username Your dataservice user name 
#' @param api_key Your API key
#' @param file_to_download The name of the file to retrieve.
#' @param target Path to the location, at which to store the file. If NULL, the file will be
#' saved to the working directory.
#' @examples
#' httr_response <- get_cached_data(userame, apikey, "some_file.xlsx")
#' get_cached_data(username, apikey, "some_file.xlsx", target="C:/data/some_other_name.xlsx")
#' @import httr
#' @export 
get_cached_data <- function(username, api_key, file_to_download, target=NULL) {
  
  # Make a listing call to check credentials
  # Otherwise the API response will be written to the target location without error
  get_cdc_files(username, api_key)
  
  url_template <- "https://datenservice.kof.ethz.ch/api/v1/user/%s/datasets/%s?apikey=%s"
  
  download_url <- sprintf(url_template, username, file_to_download, api_key)
  
  if(is.null(target)) {
    target <- file_to_download
  }
  
  GET(download_url, write_disk(target, overwrite=TRUE))
}

File: python_package/kofdata/get_cached_data.py
from errors import KofdataError
from helpers import get_cdc_files
from requests import get
import os

def get_cached_data(username, api_key, file, target=None):
    # Check credentials by making listing call...
	get_cdc_files(username, api_key)
	
	if( target is None ):
		target = os.path.join(os.getcwd(), file)
	
	url = 'https://datenservice.kof.ethz.ch/api/v1/user/{}/datasets/{}?apikey={}'.format(username, file, api_key)
	
	f = get(url, stream=True)
	
	print(f.status_code)
	
	if(f.status_code == 200):
		with open(target, 'wb') as fd:
			for chunk in f.iter_content(chunk_size=128):
				fd.write(chunk)
	elif(f.status_code == 404):
		raise KofdataError('Could not find file "{}"!'.format(file))
	#else:
		# Raise generic exception